pandas/tseries/offsets.py

```diff
@@ -1328,20 +1346,24 @@ def _end_apply_index(self, dtindex):
 
         base, mult = libfrequencies.get_freq_code(self.freqstr)
         base_period = dtindex.to_period(base)
+        if not isinstance(base_period._data, np.ndarray):
```

Comment:
you seem to be doing this unwrappng a lot, wouldn't it be better to have a ``._to_unwrapped_period(..)``? that just does this (basically to_period() with unwrapping to the base data)

Reply:
The issue here is that we want to unwrap a PeriodIndex, but leave a PeriodArray unchanged.  Or are you suggesting that `_to_unwrapped_period()` would return `self` for the PeriodArray?  I'm kind of expecting that down the road we'll have some general version of this to allow us to treat EAs and EA-backed Indexes symmetrically.

Reply:
just a thought. happy to defer, but i think it does make sense to have methods that unwrap on Index but return self for Array